incubator/hnc/pkg/validators/hierarchy.go

```diff
@@ -291,6 +299,31 @@ func isHNCServiceAccount(user *authnv1.UserInfo) bool {
 	return false
 }
 
+// checkConfig is checking whether namespaces in the hierarchy configuration meet kubernetes requirements.
+// if required children's field contains invalid name, it returns admission response that reject namespaces creation.
+func checkConfig(hc *api.HierarchyConfiguration) admission.Response {
+
+	// Check if children names in requiredChildren field obey kubernetes namespace regex format.
+	// invalidRCs accomodates illegal required child(RC) name(s).
+	invalidRCs := []string{}
+	for _, rc := range hc.Spec.RequiredChildren {
+		if resp := validateNamespace(rc); resp != nil {
+			invalidRCs = append(invalidRCs, rc)
+		}
+	}
+
+	if len(invalidRCs) > 0 {
+		return deny(metav1.StatusReasonBadRequest, fmt.Sprintf("The following required children are not valid namespace names: %v", strings.Join(invalidRCs, " ")))
```

Comment:
Please use `%s` instead of `%v`; in this case they're identical but we should use the strongest type we can.

Also, maybe consider `strings.Join(invalidRCs, ", ")` - the commas will probably make the list easier to read.

Reply:
ok